airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -102,66 +141,90 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
+    """
+
+    primary_key = "reportId"
+    path_prefix = f"/reports/{REPORTS_API_VERSION}/reports"
+    wait_seconds = 30
+
+    def should_retry(self, response: requests.Response) -> bool:
+        should_retry = super().should_retry(response)
+        if not should_retry:
+            should_retry = response.json().get(self.data_field, {}).get("processingStatus") in ["IN_QUEUE", "IN_PROGRESS"]
+        return should_retry
+
+    def backoff_time(self, response: requests.Response):
+        return self.wait_seconds
+
+    def _create_report(self, stream_state: Mapping[str, Any] = None) -> Mapping[str, Any]:
+        request_headers = self.request_headers(stream_state=stream_state)
+        report_data = {"reportType": self.name, "marketplaceIds": self.marketplace_ids}
+        create_report_request = self._create_prepared_request(
+            method="POST",
+            path=self.path_prefix,
+            headers=dict(request_headers, **self.authenticator.get_auth_header()),
+            data=json_lib.dumps(report_data),
         )
+        report_response = self._send_request(create_report_request, {})
 
-    def request_headers(self, *args, **kwargs) -> Mapping[str, Any]:
-        return {"content-type": "application/json"}
+        time.sleep(self.wait_seconds)
 
+        return report_response.json()[self.data_field]
 
-class ReportsBase(AmazonSPStream, ABC):
-    primary_key = "reportId"
-    cursor_field = "createdTime"
-    replication_start_date_field = "createdSince"
-    next_page_token_field = "nextToken"
-    page_size_field = "pageSize"
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]):
+        return self._create_report(current_stream_state)
 
-    def path(self, **kwargs):
-        return f"/reports/{REPORTS_API_VERSION}/reports"
+    def stream_slices(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        return [self._create_report(stream_state)]
 
-    def request_params(
-        self, stream_state: Mapping[str, Any], next_page_token: Mapping[str, Any] = None, **kwargs
-    ) -> MutableMapping[str, Any]:
-        params = super().request_params(stream_state, next_page_token, **kwargs)
-        if not next_page_token:
-            params.update({"reportTypes": self.name})
-        return params
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs):
+        return f"{self.path_prefix}/{stream_slice[self.primary_key]}"
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        yield response.json().get(self.data_field, {})
 
 
-class MerchantListingsReports(ReportsBase):
+class MerchantListingsReports(ReportsAmazonSPStream):
```

Comment:
isn't this supposed to declare a cursor field? otherwise this stream will not be considered incremental